File: NetCore.Angular.WeatherWidget/NetCore.Angular.WeatherWidget/Models/WeatherModel.cs
namespace NetCore.Angular.WeatherWidget.Models
{
    public class WeatherModel
    {
        public WeatherDescriptionModel WeatherDescription { get; set; }
        public EnvironmentModel Environment { get; set; }
        public WindModel Wind { get; set; }
    }

    public class WeatherDescriptionModel
    {
        public string Main;
        public string Description;
        public string Icon = "";
    }

    public class EnvironmentModel
    {
        public double Temp;
        public int Pressure;
        public int Humidity;
        public int Visibility;
        public int Clouds;
    }

    public class WindModel
    {
        public double Speed;
        public double Deg;
        public double Gust;
    }
}

File: NetCore.Angular.WeatherWidget/NetCore.Angular.WeatherWidget/ClientApp/src/app/models/weather.model.ts
export class WeatherModel {
    weatherDescription: WeatherDescription = new WeatherDescription();
    environment: Environment = new Environment();
    wind: Wind = new Wind();
}

export class WeatherDescription {
    main: string;
    description: string;
    icon: string;
}

export class Environment {
    temp: number;
    pressure: number;
    humidity: number;
    visibility: number;
    clouds: number;
}

export class Wind {
    speed: number;
    deg: number;
    gust: number;
}
